openbas-api/src/main/java/io/openbas/service/ScenarioService.java

```diff
@@ -296,6 +296,12 @@ public Scenario scenario(@NotBlank final String scenarioId) {
         .orElseThrow(() -> new ElementNotFoundException("Scenario not found"));
   }
 
+  public Scenario scenarioFromExerciseId(@NotBlank final String exerciseId) {
+    return this.scenarioRepository
+        .findByExercises_Id(exerciseId)
+        .orElseThrow(() -> new ElementNotFoundException("Scenario not found"));
```

Comment:
Suggestion: "Scenario not found" => "Scenario not found for simulation " + simulationId